tests/src/FunctionalJavascript/CollaboraIframeFormatterTest.php

```diff
@@ -0,0 +1,120 @@
+use Drupal\FunctionalJavascriptTests\WebDriverTestBase;
+use Drupal\Tests\collabora_online\Traits\MediaFormatterTrait;
+use Drupal\Tests\collabora_online\Traits\TestDocumentTrait;
+
+/**
+ * @coversDefaultClass \Drupal\collabora_online\Plugin\Field\FieldFormatter\CollaboraPreviewEmbed
+ */
+class CollaboraIframeFormatterTest extends WebDriverTestBase {
+
+  use MediaFormatterTrait;
+  use TestDocumentTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'collabora_online',
+    'collabora_online_test',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $defaultTheme = 'stark';
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    $this->enableStandaloneMediaPage();
+
+    // Set a more distinctive field name.
+    FieldConfig::load('media.document.field_media_file')
+      ->set('label', 'Field with attached file')
+      ->save();
+
+    $this->setFormatter('collabora_preview_iframe', [
+      'aspect_ratio' => '5 / 2',
+    ]);
+
+    $this->createTestDocument();
+  }
+
+  /**
+   * Tests the display of the iframe formatter.
+   */
+  public function testIframeFormatterDisplay(): void {
+    // A user with insufficient permissions can see the media, but not the
+    // preview button.
+    $account = $this->drupalCreateUser([
+      'access content',
+    ]);
+    $this->drupalLogin($account);
+    $this->visitMediaPage();
+    $assert_session = $this->assertSession();
+    $assert_session->pageTextNotContains('Field with attached file');
+    $assert_session->elementNotExists('css', 'iframe');
+
+    // A user with sufficient permissions can see the preview button.
+    $account = $this->drupalCreateUser([
+      'access content',
+      'preview document in collabora',
+    ]);
+    $this->drupalLogin($account);
+    $this->visitMediaPage();
+    $assert_session->pageTextContains('Field with attached file');
+    $iframe = $assert_session->elementExists('css', 'iframe.cool-iframe');
+
+    // Assert iframe attributes and dimensions.
+    $this->assertSame('/cool/view/' . $this->media->id(), $iframe->getAttribute('src'));
+    $this->assertSame('aspect-ratio: 5 / 2', $iframe->getAttribute('style'));
+    $this->assertIframeDimensions(1000);
+  }
+
+  /**
+   * Asserts iframe dimensions and position.
+   *
+   * This verifies that the CSS is correctly applied.
+   *
+   * @param int $expected_width
+   *   Expected width of the iframe.
+   */
+  protected function assertIframeDimensions(int $expected_width): void {
+    [$iframe_box, $parent_box] = $this->getSession()->evaluateScript(<<<JS
+[
+  document.querySelector('iframe').getBoundingClientRect(),
+  document.querySelector('iframe').parentElement.getBoundingClientRect(),
+]
+JS);
+
+    // The iframe size and position is exactly as its parent element.
+    $this->assertSame($parent_box, $iframe_box);
+
+    // The iframe width is as expected.
+    $this->assertSame($expected_width, $iframe_box['width']);
```

Comment:
I was considering to drop this width assertion.\
It tells more about the 'stark' theme and the current viewport dimensions than about the formatter.

One thing I want to verify here is that the entire thing does not shrink away. But the parent element is controlled by the theme, not by us, so it should always have full content width.

I want the test to fail if we try to improve or refactor something in the future which breaks the appearance.

In an earlier version I was also testing this in different themes and with different viewport width.
But I think for this iframe/embed formatter this is not relevant.

Reply:
I don't think indeed that it's common practice to test these elements. The ratio yes as it's generated by us, but testing the css width not really.

Reply:
I removed the width assertion.